### ISSUE TITLE
Gradient Descend in Python [HACKTOBERFEST]

### ISSUE THREAD

**Issue by Kumar-laxmi** [Kumar-laxmi/Algorithms#1785]
Gradient Descent is an iterative algorithm that is used to minimize a function by finding the optimal parameters. Gradient Descent can be applied to any dimension function i.e. 1-D, 2-D, 3-D. In this article, we will be working on finding global minima for parabolic function (2-D) and will be implementing gradient descent in python to find the optimal parameters for the linear regression equation (1-D). Before diving into the implementation part, let us make sure the set of parameters required to implement the gradient descent algorithm. To implement a gradient descent algorithm, we require a cost function that needs to be minimized, the number of iterations, a learning rate to determine the step size at each iteration while moving towards the minimum, partial derivatives for weight & bias to update the parameters at each iteration, and a prediction function. 

![image](https://github.com/Kumar-laxmi/Algorithms/assets/76027425/b5f6123f-9da7-4a98-bb5c-7573529c2c95)
